contracts/utils/SafeT.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.9;
+
+import "../token/T.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+/// @notice A wrapper around OpenZeppelin's `SafeERC20Upgradeable` but for
+///         the T token. Motivation is to prevent upgradeable contracts using T
+///         from depending on the `Address` library, which can be problematic
+///         since it uses `delegatecall`, which is discouraged by OpenZeppelin.
```

Comment:
Thanks for the clarification, I will add it. BTW, I think OpenZeppelin's plug-in may be over-reacting, since even though we use the Address and SafeERC20 libraries, we don't use any of their methods that make use of delegatecalls, as far as I'm aware. So really what we do here is more like a precaution, rather than a real mitigation.

Reply:
✅